docs/import/add_disaster.js

```diff
@@ -60,14 +76,235 @@ function enableWhenReady() {
+  // starts at 0
+  for (let i = 1; i <= numLayers; i++) {
+    $('#tbody tr:nth-child(' + i + ') .index-td').html(numLayers - i);
+  }
+
+  return updateLayersInFirestore();
+}
+
+/**
+ * Utility method for creating table divs with a given inner text
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function createTd() {
+  return $(document.createElement('td'));
+}
+
+/**
+ * Add text to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withText(td, layer, property) {
+  return td.html(layer[property]);
+}
+
+/**
+ * Add a sample of info from a list to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withList(td, layer, property) {
+  return td.html(layer[property][0]);
+}
+
+const layerTypeStrings = new Map();
+for (const t in LayerType) {
+  if (LayerType.hasOwnProperty(t)) {
+    layerTypeStrings.set(LayerType[t], t);
+  }
+}
+
+/**
+ * Add layer type info to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+}
+ */
+function withType(td, layer, property) {
+  return td.html(layerTypeStrings.get((layer[property])));
+}
+
+/**
+ * Add checkbox capabilities to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLElement> }
+ */
+function withCheckbox(td, layer, property) {
+  const checkbox = $(document.createElement('input'))
+                       .prop('type', 'checkbox')
+                       .prop('checked', layer[property])
+                       .on('change', (event) => {
+                         const index = $('#tbody > tr').length -
+                             $('tr').index($(event.target).parents('tr'));
+                         getCurrentLayers()[index][property] =
+                             checkbox.is(':checked');
+                         return updateLayersInFirestore();
+                       });
+  return $(td).append(checkbox);
+}
+
+/**
+ * Create an instance of the color boxes for the color col.
+ * @param {string} color what color to make the box.
+ * @return {JQuery<HTMLDivElement>}
+ */
+function createColorBox(color) {
+  return $(document.createElement('div'))
+      .addClass('box')
+      .css('background-color', color);
+}
+
+/**
+ * Add color function info to the given td.
+ * @param {JQuery<HTMLElement>} td
+ * @param {Object} layer
+ * @param {string} property
+ * @param {number} index
+ * @return {JQuery<HTMLElement>}
+ */
+function withColor(td, layer, property, index) {
+  const colorFunction = layer[property];
+  if (!colorFunction) {
+    td.html('N/A').addClass('na');
```

Comment:
I think we'll appreciate the consistency of keeping all data at the same HTML hierarchical level, so this N/A and class na would go on a span that was in the td, rather than at the td level. But we can change that in the next PR if you prefer (basically, as soon as you make this editable, I think it will make more sense to have the current status in a separate element that you can easily detach). My recommendation would be to have a single object inside the td. Either a color box, or a div containing multiple color boxes, or a div containing text. Then at the end of this function you append whatever element it is to the td. But my main concern is less the code duplication of td.append and more the issue of cleanly modifying this td without needing to check on what it's current state is.

Also, is it really true that it's N/A in this case? What if this is definitely a type that requires a color function, but the user hasn't entered it yet? I think we're going to want the user to be able to add a FeatureCollection without making them necessarily specify the color right away. Or are you envisioning that we always populate a default color?